titus-common/src/main/java/com/netflix/titus/common/util/feature/FeatureRollout.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.common.util.feature;
+
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation that defines the strategy for rolling out of a new feature.
+ */
+@Retention(RUNTIME)
+@Target({FIELD, METHOD})
+public @interface FeatureRollout {
+
+    /**
+     * Feature unique identifier.
+     */
+    String featureId();
+
+    /**
+     * Rollout deadline date.
+     */
+    String deadline();
```

Comment:
should this be a `java.time.LocalDateTime` (or similar) so we avoid dealing with string formatting/parsing issues here?

Reply:
Java annotation value types are very restricted: https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.6.1

Reply:
ah, makes sense! Carry on